### DISCUSSION TITLE
Combine Ref with Computed

### DISCUSSION THREAD

**Discussion by Nemure231** [vue-macros/vue-macros#400] [Ideas]
Hi, first thing thanks for this plugin, its really help me a lot. Anyway, i have an idea, but cant really think how to implement it. The motive is related to Pinia, i have to write ref and computed to create writeable global data for pinia state.
```JS
export const useCounter = defineStore('counter', () => {
  const countState = ref(0)

  const count = computed({
    get: () => countState.value,
    set: (val) => countState.value = val
  })
  return { count }
})

```
Is it possible that vue macros can make a shorthand for ref and computed? maybe a simple shorthand for ref + computed would be cool, the end result of the idea maybe like this:

```JS
const count = computedRef(() => 0)
```

Also tell me if this already exists in somewhere.

**Comment by sxzz** [comment]
Why don't return a Ref directly?
```ts
export const useCounter = defineStore('counter', () => {
  const count = ref(0)

  return { count }
})
```

**Reply by Nemure231** [reply]
Its throw an error, about ref is not writeable, so i need to combine ref with computed...

**Reply by sxzz** [reply]
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgEwKYDNgDtUGUbSpwC+c6UEIcA5GNsAIbUBQoksicUGJZFV1AG4BXVC2aoAHu3gBjCFgDO8YYtQBhCMKwxUUOAF4UGbHgLcAFNXnbdUagBo4FgJSGAfImZw48pXK0dQy4MCwAGF2ZvEJhhKCxOGyDiZmJI5lUNQLtXADokmCA

Could you please provide the docs about your usage?

**Reply by Nemure231** [reply]
hi, i visit pinia again and they now support ref directly.... you are right tho, i forgot to read it like 1 year, and still using their old way to updating the global state until now. anyway thanks for your fast reply.
